File: frontend/src/types/bindImport.ts
import type { DnsRecordFormValues } from "./dnsRecord";

export interface ParsedRecordPreview {
  line_number: number;
  raw_line: string;
  record: DnsRecordFormValues | null;
  error: string | null;
}

export interface BindImportPreviewResponse {
  zone_id: number;
  total_lines_parsed: number;
  valid_count: number;
  invalid_count: number;
  preview: ParsedRecordPreview[];
  import_token: string;
}

export interface BindImportSummary {
  created: number;
  skipped: number;
  failed: number;
  failures: string[];
}

File: backend/app/schemas/bind_import.py
from __future__ import annotations

from pydantic import BaseModel

from app.schemas.dns_record import DnsRecordCreate


class ParsedRecordPreview(BaseModel):
    line_number: int
    raw_line: str
    record: DnsRecordCreate | None = None
    error: str | None = None


class BindImportPreviewResponse(BaseModel):
    zone_id: int
    total_lines_parsed: int
    valid_count: int
    invalid_count: int
    preview: list[ParsedRecordPreview]
    import_token: str


class BindImportConfirmRequest(BaseModel):
    import_token: str
    skip_invalid: bool = True


class BindImportSummary(BaseModel):
    created: int
    skipped: int
    failed: int
    failures: list[str] = []
